app/controllers/resources_controller.rb

```diff
@@ -49,8 +49,10 @@ def cocina_model
     Cocina::Models::RequestDRO.new(model_params)
   end
 
-  # JSON-API error response. See https://jsonapi.org/
+  # JSON-API error response. See https://jsonapi.org/.
   def build_error(msg, err, code, status)
+    m = err.message.match(/:\s(\d{3})/)
```

Comment:
Using a regex based on the return value in `dor-services-client` ResponseErrorFormatter. More involved solution would be to add the error code as a variable to `Dor::Services::Client::UnexpectedResponse`, `Dor::Services::Client::ConnectionFailed`, and maybe `Dor::WorkflowException`